### ISSUE TITLE
Minor issues

### ISSUE THREAD

**Issue by GasperPaul** [AndyLanYT/hangman-game#1]
Generally, very good.

https://github.com/AndyLanYT/hangman-game/blob/ad635db031ff520c0f3f09c0b44b7334f07ca83e/hangman%20game%20v2.3/hangman%20v2.3.py#L115-L119

This is right, but a bit more robust choice would be to use `bool` value instead of a number, like:
```python
counter = True
for letter in secret_word:
    counter = counter and letter in letters_guessed
```

Also, you can use `map` or `all` functions.

https://github.com/AndyLanYT/hangman-game/blob/ad635db031ff520c0f3f09c0b44b7334f07ca83e/hangman%20game%20v2.3/hangman%20v2.3.py#L132

You should use `'_ '`, not `'_'` (note the space) as a wildcard. Same goes for `match_with_gaps`. This violates 1B.

https://github.com/AndyLanYT/hangman-game/blob/ad635db031ff520c0f3f09c0b44b7334f07ca83e/hangman%20game%20v2.3/hangman%20v2.3.py#L136

You should do this only once, in line 132.

https://github.com/AndyLanYT/hangman-game/blob/ad635db031ff520c0f3f09c0b44b7334f07ca83e/hangman%20game%20v2.3/hangman%20v2.3.py#L136

This is literally `string.ascii_lowercase`. But I know you don't like to use other modules.

https://github.com/AndyLanYT/hangman-game/blob/ad635db031ff520c0f3f09c0b44b7334f07ca83e/hangman%20game%20v2.3/hangman%20v2.3.py#L220-L224
https://github.com/AndyLanYT/hangman-game/blob/ad635db031ff520c0f3f09c0b44b7334f07ca83e/hangman%20game%20v2.3/hangman%20v2.3.py#L231-L235

This should be a separate function to prevent code duplication.

Also, you should not reset warnings. This is not part of the specification 2.D.2.

https://github.com/AndyLanYT/hangman-game/blob/ad635db031ff520c0f3f09c0b44b7334f07ca83e/hangman%20game%20v2.3/hangman%20v2.3.py#L250

You should subtract 2 if this was a vowel (`aouei`). This violates 2.D.6.

https://github.com/AndyLanYT/hangman-game/blob/ad635db031ff520c0f3f09c0b44b7334f07ca83e/hangman%20game%20v2.3/hangman%20v2.3.py#L295-L303

You'd have `get_available_letters` for this if you've implemented it right. It should return this value.

https://github.com/AndyLanYT/hangman-game/blob/ad635db031ff520c0f3f09c0b44b7334f07ca83e/hangman%20game%20v2.3/hangman%20v2.3.py#L263-L266

You don't display player's score. This violates 2.E.3.

https://github.com/AndyLanYT/hangman-game/blob/ad635db031ff520c0f3f09c0b44b7334f07ca83e/hangman%20game%20v2.3/hangman%20v2.3.py#L295-L303

The logic here is right. But the usage of `counter` is unnecessary. Because this is a strickt rule the first mismatch will cause this to return false. This means that both `list(my_word)[i] == letters[i]` and `list(my_word)[i] == '_' and letters[i] in list(my_word)` should return false, if they are true. Otherwise, there's a match, we do nothing and go to the next letter. If we never exited with the afforementioned conditions, then we should return true.
